### PR TITLE
Add relative file name type

### PR DESCRIPTION
**Pull request by loganharbour**
Refs #26947

Needed in support of https://github.com/idaholab/moose/pull/26945, in particular to support zapdos.

### CODE REVIEW

File: framework/src/parser/Builder.C [hunk 1]
```diff
@@ -1210,9 +1212,6 @@ Builder::extractParams(const std::string & prefix, InputParameters & p)
         setvectorvector(BoundaryID);
         setvectorvector(Point);
         setvectorvector(string);
-        setvectorvector(FileName);
-        setvectorvector(FileNameNoExtension);
-        setvectorvector(MeshFileName);
```
Comment (GiudGiud) [inline framework/src/parser/Builder.C:1215]: the issue with removing those is that the person trying to add a vector of vector is not going to understand why it fails until they reach out to one of like 4 people who know about this

**Review by GiudGiud — DISMISSED** [review]

File: framework/src/parser/Builder.C (hunk 1 above)
Comment (loganharbour) [reply framework/src/parser/Builder.C:1215]: The thing is it's not supported now anyway; the proper backend stuff isn't setup for those parameters. And you do get an error that says it's not supported

Comment (loganharbour) [reply framework/src/parser/Builder.C:1215]: We could likely make this a compile time error when someone tries to add a parameter that isn't supported

Comment (GiudGiud) [reply framework/src/parser/Builder.C:1215]: sounds good

**Review by cticenhour — APPROVED** [review]

**Pull request merged**